### ISSUE TITLE
Cannot override styles using `styled()` wrapper in `v1.4.0-beta9` without `transform-template-literals`

### ISSUE THREAD

**Issue by OrcaXS** [callstack/linaria#611]
## Environment

- Linaria version: `1.4.0-beta9`
- Bundler (+ version): `webpack 4.43.0`
- Node.js version: `v12.16.3`
- OS: `macOS 10.15.4`
- React: `16.13.1`

## Description
```json
{
    "presets": [
        ["@babel/preset-env", {
            "debug": true,
            "useBuiltIns": false,
            "exclude": [
                "transform-template-literals"
            ]
        }],
        "@babel/preset-react",
        "linaria/babel"
    ]
}
```

```jsx
import React from 'react';
import { styled } from 'linaria/react';

const Wrapper = styled.div`
  width: 1em;
`
const NewWrapper = styled(Wrapper)`
  width: 2em;
`
```
If `transform-template-literals` plugin in `@babel/preset-env` is excluded, running `yarn babel test.jsx` throws:
 `Error: Using the "styled" tag in runtime is not supported. Make sure you have set up the Babel plugin correctly. See https://github.com/callstack/linaria#setup`. 

This also happens in `v2.0.0-alpha.4`.
I can confirm `v1.3.3` is working even without `transform-template-literals` plugin.

Plus, if newer target e.g. `last 1 year` is specified as `browserlist` config, 
`transform-template-literals` is not enabled by default, and the user will get `runtime is not supported` error which is very confusing. If `transform-template-literals` becomes mandatory in newer versions, I think we should document it somewhere. Thanks.

## Reproducible Demo
https://github.com/OrcaXS/linaria-issue-reproduce

**Comment by jayu** [comment]
Lol I just addressed this bug before you reported it 😂

**Comment by jayu** [comment]
closed via #609
